api/v1beta1/azuremachine_validation.go

```diff
@@ -378,3 +392,37 @@ func ValidateDiagnostics(diagnostics *Diagnostics, fieldPath *field.Path) field.
 
 	return allErrs
 }
+
+// ValidateConfidentialCompute validates the configuration options when the machine is a Confidential VM.
+func ValidateConfidentialCompute(managedDisk *ManagedDiskParameters, profile *SecurityProfile, fieldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	var securityEncryptionType SecurityEncryptionType
+
+	if managedDisk != nil && managedDisk.SecurityProfile != nil {
+		securityEncryptionType = managedDisk.SecurityProfile.SecurityEncryptionType
+	}
+
+	if profile != nil && securityEncryptionType != "" {
+		if profile.EncryptionAtHost != nil && *profile.EncryptionAtHost && securityEncryptionType == SecurityEncryptionTypeDiskWithVMGuestState {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("EncryptionAtHost"), profile.EncryptionAtHost,
+				"EncryptionAtHost cannot be set to 'true' when securityEncryptionType is set to DiskWithVMGuestState"))
+		}
+
+		if profile.VirtualizedTrustedPlatformModule == "" || profile.VirtualizedTrustedPlatformModule == VirtualizedTrustedPlatformModulePolicyDisabled {
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("VirtualizedTrustedPlatformModule"), profile.VirtualizedTrustedPlatformModule,
+				fmt.Sprintf("VirtualizedTrustedPlatformModule should be set to '%s' when securityEncryptionType is defined",
+					VirtualizedTrustedPlatformModulePolicyEnabled)))
+		}
+
+		if securityEncryptionType == SecurityEncryptionTypeDiskWithVMGuestState {
```

Comment:
are you sure DiskWithVMGuestState is what you want? I've only used CVM with VMGuestStateOnly

either way, VMGuestStateOnly should be a valid setting for CVM, I think, and works with secure boot.